### PR TITLE
feat(route): add 秀动网 (#12489)

### LINKED ISSUE #12489: 秀动票务网站RSS提案
### 类型

购物

### 网站地址

https://www.showstart.com/

### 网站描述

音乐现场售票平台

### 需要生成什么内容？

- 可以根据城市抓取演出资讯
- 可以根据风格 场地 时间 艺人 来筛选

跟大麦网这类票务网站应该比较相似

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by lchtao26**
## 该 PR 相关 Issue / Involved Issue

Close #12489

## 完整路由地址 / Example for the Proposed Route(s)

```routes
/showstart/event/571
/showstart/event/571/3
/showstart/search/live
```

## 新 RSS 路由检查表 / New RSS Route Checklist

- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/advanced/script-standard) Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 full article
  - [ ] 使用缓存 Use cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/advanced/pub-date) [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date)
  - [ ] 可以解析 Parsed
  - [ ] 时区正确 Correct time zone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

### CODE REVIEW

File: lib/v2/showstart/service.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            showStyle: showStyle || '',
+            keyword: keyword || '',
+        },
+        {
+            skipEmptyString: true,
+        }
+    )}`;
+
+    const respond = await got.get(url);
+    const $ = cheerio.load(respond.data);
```
Comment (TonyRL) [inline lib/v2/showstart/service.js:22]: Please use the POST API from `/api/web/activity/list` instead.

File: lib/v2/showstart/service.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    return {
+        $,
+        data: {
+            title: '秀动网',
+            link: host,
+            item: items,
+        },
+    };
+}
```
Comment (TonyRL) [inline lib/v2/showstart/service.js:40]: Please use their web API at `/api/web/activity/list` instead.

<details><summary>Quick Snippets</summary>

<details><summary>event.js</summary>

```js
const { post, getAccessToken } = require('./utils');

module.exports = async (ctx) => {
    const accessToken = await getAccessToken();

    const result = await post('/web/activity/list', accessToken, {
        pageNo: '1',
        pageSize: '20',
        cityCode: '', // ctx.params.cityCode
        activityIds: '',
        coupon: '',
        keyword: '',
        organizerId: '',
        performerId: '',
        showStyle: '',
        showTime: '',
        showType: '',
        siteId: '',
        sortType: '',
        themeId: '',
        timeRange: '',
        tourId: '',
        type: '',
        tag: '',
    });

    console.log(JSON.stringify(result));
};
```
</details>

<details><summary>utils.js</summary>

```js
const got = require('@/utils/got');
const md5 = require('@/utils/md5');

const uuid = (length = 20) => {
    const e = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz' + Date.now();
    const r = [];
    for (let i = 0; i < length; i++) {
        r.push(e.charAt(Math.floor(Math.random() * e.length)));
    }
    return r.join('');
};

const cookieMap = new Map([['token', uuid(32).toLowerCase()]]);

const devioceInfo = {
    vendorName: '',
    deviceMode: '',
    deviceName: '',
    systemName: '',
    systemVersion: '',
    cpuMode: ' ', // Note the space
    cpuCores: '',
    cpuArch: '',
    memerySize: '',
    diskSize: '',
    network: '',
    resolution: '1920*1080',
    pixelResolution: '',
};

const getAccessToken = async () => {
    const { result } = await post('/waf/gettoken');
    cookieMap.set('accessToken', result.accessToken.access_token);
    cookieMap.set('idToken', result.idToken.id_token);
    return cookieMap.get('accessToken');
};

const post = async (requestPath, accessToken = md5(Date.now().toString()), payload) => {
    const traceId = uuid(32) + Date.now();

    const { data: response } = await got.post(`https://www.showstart.com/api${requestPath}`, {
        headers: {
            cdeviceinfo: encodeURIComponent(JSON.stringify(devioceInfo)),
            cdeviceno: cookieMap.get('token'),
            cookie: [...cookieMap.entries()].map(([key, value]) => `${key}=${value}`).join('; '),
            crpsign: md5(accessToken + 'web' + cookieMap.get('token') + (payload ? JSON.stringify(payload) : '') + requestPath + '999web' + traceId),
            crtraceid: traceId,
            csappid: 'web',
            cterminal: 'web',
            cusat: accessToken,
            cusid: '',
            cusit: '',
            cusname: '',
            cusut: '',
            cversion: '999',
        },
        json: payload,
    });

    return response;
};

module.exports = {
    post,
    getAccessToken,
    uuid,
};
```
</details>

</details>

Comment (TonyRL) [reply lib/v2/showstart/service.js:40]: You can also obtain a mapping of `cityCode`, `showStyle` using
```js
await post('/web/activity/list/params', accessToken)
```

Comment (lchtao26) [reply lib/v2/showstart/service.js:40]: Thanks a lot, It's really helpful, I will change the code later.

**Pull request merged**